src/odemis/acq/align/z_localization.py

```diff
@@ -0,0 +1,193 @@
+# -*- coding: utf-8 -*-
+"""
+Created on 23 March 2022
+
+@author: Kornee Kleijwegt, Daan Boltje
+
+Copyright © 2022 Kornee Kleijwegt, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms
+of the GNU General Public License version 2 as published by the Free Software
+Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+Odemis. If not, see http://www.gnu.org/licenses/.
+"""
+import logging
+import numpy
+
+try:
+    from skimage import io, exposure
+    from scipy.optimize import fmin_cg
+    import psf_extractor
+
+except ImportError as exp:
+    logging.warning("psf_extractor module and required libraries cannot be loaded."
+                    "The function determine_z_position will not work.\n"
+                    "The following exception was raised: \n %s", exp)
+    psf_extractor = None
+
+
+def huang(z, calibration_data):
+    """
+    Function for the expected features size in x/y direction in terms of z position for a certain degree of astigmatism.
+    This formula is based on the Huang algorithm as referenced below. The calibration data must be
+    made on this degree of astigmatism.
+
+    :param z (float): z position of the feature
```

Comment:
There is part of the code where you pass a numpy array instead of a single value. Please indicate that this is allowed.